pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -185,6 +185,22 @@ func MemberService(sdc *scyllav1alpha1.ScyllaDBDatacenter, rackName, name string
 		svc.Spec.ExternalTrafficPolicy = getValueOrDefault(ns.ExternalTrafficPolicy, "")
 	}
 
+	rackSpec, _, ok := slices.Find(sdc.Spec.Racks, func(rs scyllav1alpha1.RackSpec) bool {
+		return rs.Name == rackName
+	})
+	if !ok {
+		return nil, fmt.Errorf("can't find rack spec having %q name", rackName)
+	}
+
+	if sdc.Spec.RackTemplate != nil {
```

Comment:
This `if` looks redundant - the nil check is repeated as the 1st instruction within `applyRackTemplateOnRackSpec`.

Reply:
yep, removed